Sources/CoreFoundation/CFPlatform.c

```diff
@@ -2274,6 +2274,23 @@ CF_EXPORT int _CFPosixSpawnFileActionsAddClose(_CFPosixSpawnFileActionsRef file_
   return posix_spawn_file_actions_addclose((posix_spawn_file_actions_t *)file_actions, filedes);
 }
 
+CF_EXPORT int _CFPosixSpawnFileActionsChdir(_CFPosixSpawnFileActionsRef file_actions, const char *path) {
+  // Standardized as posix_spawn_file_actions_addchdir in POSIX.1-2024 (June 2024)
+  // _np variant available in:
+  //  - Solaris 11.3 (October 2015)
+  //  - Glibc 2.29 (February 2019)
+  //  - macOS 10.15 (October 2019)
+  //  - musl 1.1.24 (October 2019)
+  //  - FreeBSD 13.1 (May 2022)
+  //  - Android 14 (October 2023)
+  // standard variant available in:
+  //  - Solaris 11.4 (August 2018)
+  //  - NetBSD 10.0 (March 2024)
+  // missing in:
+  //  - OpenBSD
+  return posix_spawn_file_actions_addchdir_np((posix_spawn_file_actions_t *)file_actions, path);
```

Comment:
Do we need to provide support for both `_np` and `standard` variant here? Otherwise this code won't compile on Solaris, NetBSD, and OpenBSD right?

Reply:
Solaris has both variants, NetBSD has only the standard variant, OpenBSD/QNX have neither, and the remaining ones I listed have only the _np variant.

So the current code will compile fine on Solaris, but you're right that NetBSD will fail. I could change it to be more like:

```c
#if Solaris || Glibc || Apple || musl || FreeBSD || Android
posix_spawn_file_actions_addchdir_np
#else 
posix_spawn_file_actions_addchdir // NetBSD
#endif
```

...so that NetBSD works, and OpenBSD and QNX will "just work" once they update their POSIX conformance, though.

Fallback if neither variants are supported... let's leave that as an exercise for future contributors working to port Swift to those platforms? We don't have autotools-style configure checks so I'm not sure we can easily test for the presence of the function right now anyways, versus a bunch of hardcoded checks for specific platform macros.

And there isn't a good way to do a fallback while retaining thread safety by not mutating the calling process's working directory.

Reply:
Ahh sounds good. That's reasonable.